addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -853,36 +853,37 @@ var SnippetEditor = Widget.extend({
      * @private
      * @param {OdooEvent} ev
      */
-    _onOptionUpdate: function (ev) {
+    _onOptionUpdate: async function (ev) {
         var self = this;
-
         // If multiple option names are given, we suppose it should not be
         // propagated to parent editor
         if (ev.data.optionNames) {
             ev.stopPropagation();
-            _.each(ev.data.optionNames, function (name) {
-                notifyForEachMatchedOption(name);
-            });
+            for (const name of ev.data.optionNames) {
+                await notifyForEachMatchedOption(name);
+            }
         }
         // If one option name is given, we suppose it should be handle by the
         // first parent editor which can do it
         if (ev.data.optionName) {
-            if (notifyForEachMatchedOption(ev.data.optionName)) {
+            const cond = await notifyForEachMatchedOption(ev.data.optionName);
+            if (cond) {
                 ev.stopPropagation();
             }
         }
```

Comment:
Those lines here cannot work :wink: You `await` something then stop the propagation of an event. By the time the awaited instruction is done, your event is long gone. I don't know if or why that would be useful... but if it was there for some reason, it is now broken.

Also: `_onOptionUpdate` is now `async`, which means that all places calling it should account for the async part when they must... this has not been done either. For example the `position` method of the `gallery_img` option. Not every function can be made async without carefully checking all the codes calling it and it's most of the time not trivial.

More fundamental question here by curiosity: why would the introduction of vDOM require more async? It's good for our long terme goal of having everything extendable the way the devs/community wants it but I don't see the reason why the _vDOM_ would need async operations?

Reply:
cc @Goaman since you wrote this fix with @sebgeelen

Reply:
It's indeed true about the await. We noticed the problem when investigating and it was obviously forgotten before merging.
The event should be stopped if there is one `snippetOptionInstance` existing (i.e. `key === name || regex.test(key)`), which can be tested without requiring the next microtask. We will provide a fix.

We'll also investigate to adapt the code that is using `_onOptionUpdate`.

The VDOC require async operation because of browser quirks, it need to setTimeout on keydown DOM event. Which require to have a mutex to insure the correct order of the events when introduced in the document.
For example:
- you hit the key `a` (will be process in next tick)
- you make an operation `X` in the VDOC without the mutex (processed directly)

Then, the sequence order is not correct:
- the operation `X`
- the key `a`

It happen as soon as there is a concurrency issue (e.g. when there is a lag in the browser enclosing two or more operations).

Reply:
see #62307